spec/inputs/gelf_spec.rb

```diff
@@ -84,9 +153,9 @@
       # using explicit and certainly useless to_f here just to leave no doubt about the numeric type involved
 
       it "should coerce and preserve millisec precision in iso8601" do
-        expect(LogStash::Inputs::Gelf.coerce_timestamp(946702800.1.to_f).to_iso8601).to eq("2000-01-01T05:00:00.100Z")
-        expect(LogStash::Inputs::Gelf.coerce_timestamp(946702800.12.to_f).to_iso8601).to eq("2000-01-01T05:00:00.120Z")
-        expect(LogStash::Inputs::Gelf.coerce_timestamp(946702800.123.to_f).to_iso8601).to eq("2000-01-01T05:00:00.123Z")
+        expect(LogStash::Inputs::Gelf.coerce_timestamp(Rational(946702800.1).truncate(3)).to_iso8601).to eq("2000-01-01T05:00:00.100Z")
+        expect(LogStash::Inputs::Gelf.coerce_timestamp(Rational(946702800.12).truncate(3)).to_iso8601).to eq("2000-01-01T05:00:00.120Z")
+        expect(LogStash::Inputs::Gelf.coerce_timestamp(Rational(946702800.123).truncate(3)).to_iso8601).to eq("2000-01-01T05:00:00.123Z")
```

Comment:
These change the _specification_, effectively removing the specification for our `coerce_timestamp` should behave when given a`Float` and replacing it with a specification for how it should behave with a`Rational`. Since in practice we are much more likely to receive a `Float` than a `Rational`, we should leave the original specification in place.

I think we need to instead change our _implementation_ to behave in the way we expect when receiving float values as input and change this spec to use the `be_a_logstash_timestamp_equivalent_to` helper to make it resilient to the serialization length:

```suggestion
        expect(LogStash::Inputs::Gelf.coerce_timestamp(946702800.1.to_f)).to be_a_logstash_timestamp_equivalent_to("2000-01-01T05:00:00.100Z")
        expect(LogStash::Inputs::Gelf.coerce_timestamp(946702800.12.to_f)).to be_a_logstash_timestamp_equivalent_to("2000-01-01T05:00:00.120Z")
        expect(LogStash::Inputs::Gelf.coerce_timestamp(946702800.123.to_f)).to be_a_logstash_timestamp_equivalent_to("2000-01-01T05:00:00.123Z")
```

Reply:
committed